### PR TITLE
[OpenTelemetry] Add optional locality label for per-call attempt metrics

### PR DESCRIPTION
**Pull request by yashykt**

### CODE REVIEW

**Comment by temawi** [comment]
In the "Weighted Round Robin LB Policy Instruments" table you have the column header as "Labels (disposition)", do you want to change that to mach the header in this table?

**Comment by yashykt** [comment]
Thanks!

File: content/en/docs/guides/opentelemetry-metrics.md [hunk 1]
```diff
@@ -43,12 +43,12 @@ Refer [A66: OpenTelemetry Metrics] for details.
...
+| grpc.client.attempt.<br>started | Counter | {attempt} | grpc.method (required), grpc.target (required) | The total number of RPC attempts started, including those that have not completed. |
+| grpc.client.attempt.<br>duration | Histogram | s | grpc.method (required), grpc.target (required), grpc.status (required), grpc.lb.locality (optional) | End-to-end time taken to complete an RPC attempt including the time it takes to pick a subchannel. |
+| grpc.client.attempt.<br>sent_total_compressed_message_size | Histogram | By | grpc.method (required), grpc.target (required), grpc.status (required), grpc.lb.locality (optional) | Total bytes (compressed but not encrypted) sent across all request messages (metadata excluded) per RPC attempt; does not include grpc or transport framing bytes. |
+| grpc.client.attempt.<br>rcvd_total_compressed_message_size | Histogram | By | grpc.method (required), grpc.target (required), grpc.lb.locality (optional) | Total bytes (compressed but not encrypted) received across all response messages (metadata excluded) per RPC attempt; does not include grpc or transport framing bytes. |
```
Comment (DNVindhya) [inline content/en/docs/guides/opentelemetry-metrics.md:51]: We also have `grpc.status` as an attribute on `grpc.client.attempt.rcvd_total_compressed_message_size`.

Comment (yashykt) [reply content/en/docs/guides/opentelemetry-metrics.md]: Done

**Review by DNVindhya — APPROVED** [review]

**Review by ejona86 — APPROVED** [review]

**Pull request merged**